### PR TITLE
Add slides to README; explain API.

### PR DESCRIPTION
**Pull request by smolkaj**
This partially addresses https://github.com/p4lang/p4-constraints/issues/18.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ The project currently provides two main artifacts:
    entries satisfy the constraints placed on their respective tables or not.
    (Note that the CLI is intended for testing and experimentation, not for
    production use.)
+   
+**_Check out [these slides](docs/2020-06-03_p4rt-wg.pdf) for a tour of p4-constraints._**
```
Comment (stefanheule) [inline README.md:21]: Maybe we could add something like "(up to date as of 2020-06-03)"

File: README.md [hunk 2]
```diff
@@ -62,6 +64,35 @@ should not require general ternary matches on the destination address; the
...
+// Checks if a given table entry satisfies the entry constraint attached to its
+// associated table. Returns true if this is the case or if no constraint
+// exists. Returns an InvalidArgument Status if the entry belongs to a table not
+// present in ConstraintInfo, or if it is inconsistent with the table definition
+// in ConstraintInfo.
+util::StatusOr<bool> EntryMeetsConstraint(const p4::v1::TableEntry& entry,
+                                          const ConstraintInfo& context);
+```
+You can learn more about use cases for p4-constraints and possible extensions in
+[our slide deck](docs/2020-06-03_p4rt-wg.pdf).
```
Comment (stefanheule) [inline README.md:94]: Maybe we should just copy the usecases here?  It's short enough and seems useful.

File: README.md [hunk 3]
```diff
@@ -62,6 +64,35 @@ should not require general ternary matches on the destination address; the
 constraint documents this intend and let's us catch accidental ternary matches
 installed by the control plane at runtime.
 
+## API
```
Comment (stefanheule) [inline README.md:67]: This is awesome, thanks for documenting this. I still feel like we should probably reoganize the files a bit and have a top-level p4constraints.h that has the main API, so people can only include and use that. WDYT? That can be a separate PR, but we should at least track it in an issue.

File: README.md (hunk 1 above)
Comment (smolkaj) [reply README.md:21]: Mh, do we need this given that the date is already on the first slide and also in the file name? Happy to add it if you think it's nonetheless useful, though.

File: README.md (hunk 3 above)
Comment (stefanheule) [reply README.md:67]: (I find the includes right now to be super random)

File: README.md [hunk 4]
```diff
@@ -62,6 +64,35 @@ should not require general ternary matches on the destination address; the
...
+
+At a high level, p4-constraint's API consists of only two functions:
```
Comment (stefanheule) [inline README.md:69]: Maybe we should end this section with: There are also other APIs for parsing, etc, but those are only needed for projects building on top of p4-constraints, not simple users of the library? This makes it sound like there is nothing else?

File: README.md (hunk 3 above)
Comment (smolkaj) [reply README.md:67]: Oh, I see. Yes, happy to do that. Not sure what the C++ conventions for this kind of stuff are.

File: README.md (hunk 4 above)
Comment (smolkaj) [reply README.md:69]: Yes, good point

File: README.md (hunk 2 above)
Comment (smolkaj) [reply README.md:94]: True. Will do.

File: README.md (hunk 3 above)
Comment (stefanheule) [reply README.md:67]: I'm also not sure, but the imports are super random, these are definitly not the files I would go look for. We should see some other libraries maybe for inspiration. But generally I think just having a single .h file that is clearly the main API file, and everything else be a bit more "hidden" seems like a good start?

File: README.md (hunk 1 above)
Comment (stefanheule) [reply README.md:21]: I think we should. This will get stale quickly, and we don't want people to think the language is much smaller than it really is if we extend it :)

File: README.md (hunk 3 above)
Comment (smolkaj) [reply README.md:67]: Yes, I agree with the sentiment.

File: README.md (hunk 1 above)
Comment (smolkaj) [reply README.md:21]: Done

File: README.md (hunk 2 above)
Comment (smolkaj) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (smolkaj) [reply README.md:67]: I looked into this briefly, and it seems that C++ has no standard mechanism for doing this kind of stuff. In particular, C++ does not support function aliases.

I would love to be wrong about this, but my general feeling is that C++ wasn't really build with modularity in mind (though hopefully modules will make it into C++20).

File: README.md (hunk 4 above)
Comment (smolkaj) [reply README.md:69]: Done.

File: README.md (hunk 3 above)
Comment (smolkaj) [reply README.md:67]: Since I don't see an immediate fix, let's leave this for another time; we already have issue #18 to track it.

**Comment by smolkaj** [comment]
Let me know if there is anything else needed before we can merge.
